internal/jobmanager/manager.go

```diff
@@ -0,0 +1,86 @@
+package jobmanager
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/aadc-dev/teleport-job-management-int/lib/jobworker"
+
+	"github.com/google/uuid"
+)
+
+// Manager handles Jobs
+type Manager struct {
+	mu   sync.RWMutex
+	jobs map[string]*jobworker.Job
+}
+
+// MakeManager makes a new instance of job manager
+func MakeManager() *Manager {
+	return &Manager{
+		jobs: make(map[string]*jobworker.Job),
+	}
+}
+
+// CreateAndStartJob creates a new job and starts it.
+// owner is the CN from the client certificate.
```

Comment:
The library is not supposed to know about about gRPC, mTLs, etc.

I would generalize this.

```suggestion
// owner is used for authorization decisions and must uniquely
// identify the owner of the job.
```

Reply:
yup thanks, will update on the next commit